data_controller.rb

```diff
@@ -0,0 +1,87 @@
+require './student'
+require './teacher'
+require './book'
+require './rental'
+require 'json'
+
+module DataController
+  def load_peoble
+    file = 'people.json'
+    data = []
+    if File.exist?(file) && !File.empty?(file)
+      JSON.parse(File.read(file)).each do |element|
+        if element['data_type'] == 'Teacher'
+          data.push(Teacher.new(id: element['id'].to_i, age: element['age'].to_i,
+                                specialization: element['specialization'], name: element['name']))
+        else
+          data.push(Student.new(id: element['id'].to_i, age: element['age'].to_i, name: element['name'],
+                                parent_permission: element['parent_permission']))
+        end
+      end
+    end
+    data
+  end
+
+  def load_books
+    file = 'books.json'
+    data = []
+    if File.exist?(file) && !File.empty?(file)
+      JSON.parse(File.read(file)).each do |element|
+        data.push(Book.new(element['title'], element['author']))
+      end
+    end
+    data
+  end
+
+  def get_person_by_id(id)
+    @persons.each { |element| return element if element.id == id.to_i }
+  end
+
+  def get_book_by_title(title)
+    @books.each { |element| return element if element.title == title }
+  end
+
+  def load_rentals
+    file = 'rentals.json'
+    data = []
+    if File.exist?(file) && !File.empty?(file)
+      JSON.parse(File.read(file)).each do |element|
+        person = get_person_by_id(element['person_id'])
+        book = get_book_by_title(element['book_title'])
+        puts book
+        data.push(Rental.new(element['date'], book, person))
+      end
+    end
+    data
+  end
+
+  def save_person
+    data = []
+    @persons.each do |person|
+      if person.instance_of?(Teacher)
+        data.push({ id: person.id, age: person.age, name: person.name,
+                    specialization: person.specialization, data_type: person.class })
```

Comment:
- [ ] As stated in the project requirement On exit from your app it should save all the data (books, people, and rentals) into JSON files. But I got the following error when i exit from the app  to store the Teacher data to the people.json So Kindly Fix the issues.
- 
![Save](https://github.com/StevenWafeek/OOP_school_library/assets/93598189/1f48c9ca-c803-4aec-b6ce-d232f681438b)